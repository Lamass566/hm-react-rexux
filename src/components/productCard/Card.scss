* {
    box-sizing: border-box;
 }
 .product-wrap {
    width: 300px;
    margin: 0 auto;
    background: white;
    padding: 0 0 20px;
    text-align: center;
    font-size: 14px;
    font-family: Lora;
    text-transform: uppercase;
    position: relative;
    
 }
 .product-item {
    position: relative;
    overflow: hidden;

 }
 .product-wrap img {
    height: 221px;
 }
 .product-buttons {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, .8);
    opacity: 0;
    transition: .3s ease-in-out;
 }
 .button {
    text-decoration: none;
    color: #c0a97a;
    font-size: 12px;
    width: 140px;
    height: 40px;
    line-height: 40px;
    position: absolute;
    top: 50%;
    left: 50%;
    border: 2px solid #c0a97a;
    transform: translate(-50%, -50%) scale(0);
    transition: .3s ease-in-out;
 }
 .product-item:hover .product-buttons {
    opacity: 1;
 }
 .product-item:hover .button {
    transform: translate(-50%, -50%) scale(1);
 }
 .button:hover {
    background: black;
 }
 .product-title {
    color: #5e5e5e;
 }
 .product-title p {
    text-decoration: none;
    color: #2e2e2e;
    font-weight: 600;
    margin: 15px 0 5px;
    padding-bottom: 7px;
    display: block;
    position: relative;
    transition: .3s ease-in-out;
 }
 .product-title p:after {
    content: "";
    position: absolute;
    width: 40px;
    height: 2px;
    background: #2e2e2e;
    left: 50%;
    margin-left: -20px;
    bottom: 0;
    transition: .3s ease-in-out;
 }
 .product-title p:hover {
    color: #c0a97a;
 }
 .product-title:hover p:after {
    background: #c0a97a;
 }
 .product-price {
    font-size: 20px;
    color: #c0a97a;
    font-weight: 700;
 }


 .flex-card-0{
   display: flex;
   flex-wrap: wrap;
   position: relative;
 }

 .flex-card-2{
   display: flex;
   flex-wrap: wrap;
   position: relative;
   z-index: -2;
 }

 .hide{
   display: none;
}
.show{
   display: block;
}



.product-flex-info{
   display: flex;
   justify-content: center;
   gap: 10px;
   margin-right: 10px;
}

.product-flex-info>button{
    padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: rgb(213, 28, 28);
  color: black;
  width: 20px;
  border-radius: 5px;
}


////

.product-wrap--table {
   width: 100%;
   display: flex;
   
}

.product-wrap--table img {
   height: 221px;
   display: flex;

}
.product-item--table {
   width: 300px;
   position: relative;
   overflow: hidden;
   display: flex;
   justify-content: center;
}

.product-title--table {
   color: #5e5e5e;
   display: flex;
   gap: 80px;
}

.product-title--table p:after {
   content: "";
   position: absolute;
   width: 40px;
   height: 2px;
   background: #2e2e2e;
   left: 50%;
   margin-left: -20px;
   bottom: 175px;
   transition: .3s ease-in-out;
}

.product-flex-info--table{
  display: flex;
  justify-content: center;
  gap: 10px;
  margin-right: 10px;
  padding-top: 14px;
}

.product-item--table:hover .product-buttons--table {
   opacity: 1;
}
.product-item--table:hover .button--table {
   transform: translate(-50%, -50%) scale(1);
}

.product-title--table p {
   text-decoration: none;
   color: #2e2e2e;
   font-weight: 600;
   margin: 15px 0 5px;
   padding-bottom: 7px;
   display: block;
   position: relative;
   transition: .3s ease-in-out;
}

.product-title--table p:hover {
   color: #c0a97a;
}
.product-title--table:hover p:after {
   background: #c0a97a;
}

.product-flex-info--table{
   display: flex;
   justify-content: center;
   gap: 10px;
   margin-right: 10px;
}

.product-flex-info--table>button{
    padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: rgb(213, 28, 28);
  color: black;
  width: 20px;
  border-radius: 5px;
  height: 30px;
}

.product-buttons--table {
   position: absolute;
   left: 0;
   top: 0;
   width: 100%;
   height: 100%;
   background: rgba(0, 0, 0, .8);
   opacity: 0;
   transition: .3s ease-in-out;
}

.product-item--table:hover .product-buttons--table {
   opacity: 1;
}
.button--table:hover {
   background: black;
}
.product-title--table {
   color: #5e5e5e;
}

.button--table {
   text-decoration: none;
   color: #c0a97a;
   font-size: 12px;
   width: 140px;
   height: 40px;
   line-height: 40px;
   position: absolute;
   top: 50%;
   left: 50%;
   border: 2px solid #c0a97a;
   transform: translate(-50%, -50%) scale(0);
   transition: .3s ease-in-out;
   padding-left: 43px;
}