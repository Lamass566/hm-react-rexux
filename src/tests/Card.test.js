import Card from "../components/productCard/card";
import * as reduxHooks from 'react-redux'
import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { useSelector } from 'react-redux';

jest.mock('react-redux');

const mockDis = jest.spyOn(reduxHooks, 'useDispatch')


describe('Card snapshot', () => {
    it('empty array', () => {
        mockDis.mockReturnValue(jest.fn())
        useSelector.mockReturnValue([])
        const p = <p>test</p>
        const component = render(<Card />)
        expect(component).toMatchSnapshot()
    })
})


describe('Card unit test', () => {
    it('include in document', () => {
        mockDis.mockReturnValue(jest.fn())
        useSelector.mockReturnValue([])
        const thisCard = {}
        const component = render(<Card/>)
        
        waitFor(() => expect(component).toBeInTheDocument());
    })
})
