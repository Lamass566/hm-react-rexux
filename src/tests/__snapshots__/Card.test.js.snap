// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot empty array 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form "
        style="background-color: red;"
      >
        <div
          class="header"
          style="background-color: darkred;"
        >
          <div
            class="flex"
          >
            Modal
            <span
              class="closeX"
            >
              X
            </span>
          </div>
        </div>
        <div
          class="body"
        >
          <p
            class="body_text"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
          </p>
          <div
            class="buttons"
          >
            <button
              class="mainButton"
              style="background-color: darkred;"
            >
              Ok
            </button>
            <button
              class="mainButton"
              style="background-color: darkred;"
            >
              Cancel
            </button>
          </div>
        </div>
      </form>
      <div
        class="product-wrap"
      >
        <div
          class="product-item"
        >
          <img
            alt="img"
          />
          <span />
        </div>
        <div
          class="product-title"
        >
          <p />
          <div
            class="product-flex-info"
          >
            <span />
            <span
              class="product-price"
            >
              $ 
            </span>
            <span />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <form
      class="form "
      style="background-color: red;"
    >
      <div
        class="header"
        style="background-color: darkred;"
      >
        <div
          class="flex"
        >
          Modal
          <span
            class="closeX"
          >
            X
          </span>
        </div>
      </div>
      <div
        class="body"
      >
        <p
          class="body_text"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
        </p>
        <div
          class="buttons"
        >
          <button
            class="mainButton"
            style="background-color: darkred;"
          >
            Ok
          </button>
          <button
            class="mainButton"
            style="background-color: darkred;"
          >
            Cancel
          </button>
        </div>
      </div>
    </form>
    <div
      class="product-wrap"
    >
      <div
        class="product-item"
      >
        <img
          alt="img"
        />
        <span />
      </div>
      <div
        class="product-title"
      >
        <p />
        <div
          class="product-flex-info"
        >
          <span />
          <span
            class="product-price"
          >
            $ 
          </span>
          <span />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
